truffleruby/src/main/c/cext/ruby.c

```diff
@@ -876,7 +876,9 @@ VALUE rb_str_new(const char *string, long length) {
 
 VALUE rb_str_new_cstr(const char *string) {
   if (truffle_is_truffle_object((VALUE) string)) {
-    return (VALUE) truffle_invoke(RUBY_CEXT, "to_ruby_string", string);
+    VALUE ruby_string = (VALUE) truffle_invoke(RUBY_CEXT, "to_ruby_string", string);
+    int len = strlen(string);
```

Comment:
No. The only case I have changed for strlen is that`StringCharPointerAdpater#GET_SIZE` will return the null terminated length so that it matches the semantics of `char *` strings, and it is that length we are interested in. I don't think it's a particularly good idea to call C functions which expected a `char *` with ruby string anyway, you're unlikely to get the answer you wanted, would certainly see inconsistencies between implementations, and should really get a warning (though I notice I didn't get one from Clang when I just tried it).